profiling/src/lib.rs

```diff
@@ -0,0 +1,713 @@
+    ZendResult::Success
+}
+
+fn get_module_version(module_name: &CStr) -> Option<String> {
+    // Safety: passing a CStr.as_ptr() will be a valid *const char.
+    let module_version = unsafe { zend::zend_get_module_version(module_name.as_ptr()) };
+    // It shouldn't be NULL as the engine calls `strlen` on it when it's
+    // registered; just being defensive.
+    if module_version.is_null() {
+        return None;
+    }
+
+    // Safety: module_version isn't null (checked above). It's also incredibly
+    // unlikely that the version string is longer than i64::MAX, so this isn't
+    // worth our time to check.
+    let cstr = unsafe { CStr::from_ptr(module_version) };
+    let version = cstr.to_string_lossy().to_string();
+    Some(version)
+}
+
+unsafe extern "C" fn startup(extension: *mut ZendExtension) -> ZendResult {
+    #[cfg(debug_assertions)]
+    trace!("startup({:p})", extension);
+
+    zend::datadog_php_profiling_startup(extension);
+
+    static ONCE: Once = Once::new();
+    ONCE.call_once(|| {
+        // Safety: CStr string is null-terminated.
+        let module_name = CStr::from_bytes_with_nul_unchecked(b"Core\0");
+        // Safety: ZEND_VERSION is written to inside a protected call_once, and
+        // the version string is defensibly copied.
+        ZEND_VERSION = get_module_version(module_name);
+
+        // Reflection uses the PHP_VERSION as its version, see:
+        // https://github.com/php/php-src/blob/PHP-8.1.4/ext/reflection/php_reflection.h#L25
+        // https://github.com/php/php-src/blob/PHP-8.1.4/ext/reflection/php_reflection.c#L7157
+        // It goes back to at least PHP 7.1:
+        // https://github.com/php/php-src/blob/PHP-7.1/ext/reflection/php_reflection.h
+        // Safety: string is null-terminated.
+        let module_name = CStr::from_bytes_with_nul_unchecked(b"Reflection\0");
+        // Safety: same as ZEND_VERSION above.
+        PHP_VERSION = get_module_version(module_name);
+    });
+
+    ZendResult::Success
+}
+
+extern "C" fn shutdown(_extension: *mut ZendExtension) {
+    #[cfg(debug_assertions)]
+    trace!("shutdown({:p})", _extension);
+}
+
+/// # Safety
+/// This is safe to call except that it cannot be called concurrently with
+/// MINIT. This _should_ be guaranteed by the PHP request model.
+#[no_mangle]
+pub unsafe extern "C" fn datadog_profiling_runtime_id() -> profiling::Uuid {
+    RUNTIME_ID
+}
+
+/// Used internally to gather time samples when the configured period has
+/// elapsed. Also used by the tracer to handle pending profiler interrupts
+/// before calling a tracing closure from an internal function hook; if this
+/// isn't done then the closure is erroneously at the top of the stack.
+///
+/// # Safety
+/// The zend_execute_data pointer should come from the engine to ensure it and
+/// its sub-objects are valid.
+#[export_name = "datadog_profiling_interrupt_function"]
+pub unsafe extern "C" fn interrupt_function(execute_data: *mut zend::zend_execute_data) {
```

Comment:
Why not name the Rust item `datadog_profiling_interrupt_function` and save a future reader grief when wondering where the C symbol came from :)